src/lib/fixed_string.hpp

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "types.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+// FixedString is a data type, which stores a string in an array of chars in order to
+// save memory space by avoiding small string optimization (SSO)
+class FixedString {
+ public:
+  // Create a FixedString from a std::string
+  // We did not find any good solution for creating a char array on stack
+  explicit FixedString(const std::string& string) : _mem(new char[string.size()]{}), _string_length(string.size()) {
+    std::memcpy(_mem, string.c_str(), _string_length);
+  }
+
+  // Create a FixedString from a memory address
+  FixedString(char* mem, size_t string_length) : _mem(mem), _string_length(string_length), _delete(false) {}
+
+  // Create a FixedString with an existing one
+  FixedString(const FixedString& other) : _mem(new char[other._string_length]{}), _string_length(other._string_length) {
+    std::memcpy(_mem, other._mem, _string_length);
+  }
+
+  FixedString(const FixedString&& other)
+      : _mem(other._delete ? other._mem : new char[other._string_length]{}), _string_length(other._string_length) {
+    std::memcpy(_mem, other._mem, _string_length);
+  }
+
+  ~FixedString() {
+    if (_delete) delete[] _mem;
+  }
+
+  // Copy assign
+  FixedString& operator=(const FixedString& other) {
+    DebugAssert(other.size() <= _string_length, "Other FixedString is longer than current maximum string length")
+        const auto copied_length = other.size() < _string_length ? other.size() : _string_length;
+    other._copy_to(_mem, copied_length);
+    // Fill unused fields of char array with null terminator
+    if (copied_length < _string_length) {
+      memset(_mem + copied_length, '\0', _string_length - copied_length);
+    }
+    return *this;
+  }
+
+  // Returns the length of the string
+  size_t size() const { return _string_length; }
```

Comment:
Would this be the case? If so, it feels quite inconsistent.
```
FixedString a("bla", 5);
a.size(); // 5
a.string().size() // 3
```

Reply:
We added a maximum_length() method which returns the maximum length that could be stored in this FixedString.

```
FixedString a("bla", 5);
a.size(); // 3
a.maximum_length(); // 5
a.string().size() // 3
```